### DISCUSSION TITLE
Splitting code into repositories for the organisation

### DISCUSSION THREAD

**Discussion by the-kaustubh** [mukta-strot/marathi-shabd#31] [Ideas]
Hey @sanketgarade, I think this might be a great time to have different repositories under the mukta-strot organisation for different purposes. For example, one for backend and one for frontend. Or just one extra repo for backend.

Let me know what do you think? Should we go for multiple repos or stick to just one?

**Comment by sanketgarade** [comment]
Ok for me. Just want to know the reasons for this. 

- need of separate repo for back end?
- also I have no idea what a back end is in the first place. 😬 And what it's purpose is. Could you please explain that in short? A link to any video or a textual explanation is also fine. 
- also what do you suggest? 2 repos - current one and extra for backend? Or 3? Current one (for what?) and one for front end and one for backend? 
- also is this splitting needed only for **search** site or it'll apply for the **home** and **browse** sites also?

**Reply by the-kaustubh** [reply]
- Need for a separate repo: Easy for maintaining, and if the repos are separate, each repo can stand for one specific purpose like one only for backend, one only having readme files for people to read and know about the project (like documentation), etc. Also it makes sense to have code splitted in multiple repo under an organisation.
- **What is backend? ** Some code needs to be run on the server side to serve pages, connect to DB, read/write database. This can't be done in user's browser.[Here's one video](https://www.youtube.com/watch?v=4r6WdaY3SOA)
- I suggest we go by functionality, 1 repo for backend on for frontend and one for all the documentation that is currently present or will be added later.
- I am suggesting for search site, so all the markdown files will go in one repo. And all the HTML, JS stuff will go in another one.

**Reply by sanketgarade** [reply]
Thanks. Understood. 

I have one concern/doubt about having separate repos (Points 3 and 4 in your comment above.) when it comes to different parts of the website which the user will interact directly, (which I understand is called as frontend).

Basically we have 3 parts -
- the home pages (documentation such readme.)
- the browse pages (static markdown pages)
- the search pages (the pages where user will enter the search words and see results)

We are going to have the frontend of all the above on 1 single repo right? Cause if we don't then the website address will be different if we put these parts in different repos. (Cause GitHub pages site addresses are locked to the repo name.) My point is that the **user must always be on the same site (web address) of Marathi-shabd no matter whether he is visiting the documentation, browse or search pages.**

**Reply by sanketgarade** [reply]
Also I made a new repo for the backend and gave you maintainer access.  I think you should be able to usual git stuff and create issues, discussions etc. Let me know if you face any issue in using that repo.

**Reply by sanketgarade** [reply]
@the-kaustubh Hi 👋 . 

Have you started working on the back end? Also do you have any tentative plan/target date to complete it?

**Reply by the-kaustubh** [reply]
Hey! Yes, I had already started working on it, the backend has the features like adding new words and fetching meanings.
There is still a lot of clean up  to do (my code is not so clean yet). I couldn't open a PR as I was busy. But soon I will be opening a PR on that repo.
a crude plan is - 
basic site in couple of days
then maybe think about login or ranking systems for meanings of words submitted by users.

**Comment by sanketgarade** [comment]
@the-kaustubh 
Looks ok to me. I'll wait for your PR. Once the basic site is ready we'll discuss details of the next step.
